lib/active_merchant/billing/gateways/globalcollect.rb

```diff
@@ -0,0 +1,294 @@
+      self.display_name = "GlobalCollect"
+      self.homepage_url = "http://www.globalcollect.com/"
+
+      self.test_url = "https://api-sandbox.globalcollect.com/"
+      self.live_url = "https://api.globalcollect.com/"
+
+      self.supported_countries = %w(AD AE AT AU BD BE BG BN CA CH CY CZ DE DK
+      EE EG ES FI FR GB GI GR HK HU ID IE IL IM IN IS IT JO KW LB LI LK LT LU
+      LV MC MT MU MV MX MY NL NO NZ OM PH PL PT QA RO SA SE SG SI SK SM TR TT
+      UM US VA VN ZA)
+      self.default_currency = "USD"
+      self.money_format = :cents
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :apiKeyId, :secretApiKey)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(money, payment, options) }
+          r.process { capture(money, r.authorization, options) }
+        end
+      end
+
+      def authorize(money, payment, options={})
+        post = nestable_hash
+        add_order(post, money, options)
+        add_payment(post, payment)
+        add_customer_data(post, options)
+        add_address(post, payment, options)
+
+        commit(:authorize, post)
+      end
+
+      def capture(money, authorization, options={})
+        post = nestable_hash
+        add_order(post, money, options)
+        add_customer_data(post, options)
+        commit(:capture, post, authorization)
+      end
+
+      def refund(money, authorization, options={})
+        post = nestable_hash
+        add_amount(post, money, options={})
+        add_refund_customer_data(post, options)
+        commit(:refund, post, authorization)
+      end
+
+      def void(authorization, options={})
+        post = nestable_hash
+        commit(:void, post, authorization)
+      end
+
+      def verify(payment, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, payment, options) }
+          r.process { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: )[^\\]*)i, '\1[FILTERED]').
+          gsub(%r(("cardNumber\\":\\")\d+), '\1[FILTERED]').
+          gsub(%r(("cvv\\":\\")\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      BRAND_MAP = {
+        "visa" => "1",
+        "american_express" => "2",
+        "master" => "3",
+        "discover" => "128"
+      }
+
+      def add_order(post, money, options)
+        post["order"]["amountOfMoney"] = {
+          "amount" => amount(money),
+          "currencyCode" => options[:currency] || currency(money)
+        }
+        post["order"]["references"] = {
+          "merchantOrderId" => options[:order_id], # Max 10 digits
+          "merchantReference" => options[:order_id],
```

Comment:
Let's do this - since order id is truncated to a pretty short length (10), let's leave it empty and put the order id only in the merchantReference field? That way there's no loss of data.

This assumes the merchantOrderId isn't required?